BCStudentSoftwareDevTeam/celts
# Add new background check type, disable save after press, and allow deletion of BG check

Issue: CELTS asked for a new background check type to be added (Student Life BG check). They also had times where the double clicked the save button so they wished for it to be disabled after they clicked it and wanted to be able to delete background checks. 
___________________________________________________________
Did: 
- Added the new background check type called Berea Student Life (BSL). 
- Added a red X button to the right of individual background check histories that, when pressed, deletes that entry from the database. 
- Made it so that the page is reloaded after a background check is submitted and made sure that the accordions were opened and focusing correctly. 
- Made it so that if a user tries to save a background check they will not be able to until both the status and date are filled in. 
- Moved Toast notifications so that they will be able to be used from anywhere in the application.
- Changed getUserBGCheckHistory so that it no longer formats the data. The formatting is now done in the HTML. 
- Refactored test_getUserBGCheckHistory to match with the new way background check history is returned. 

___________________________________________________________
To test:
- Checkout the branch add-and-enhance-background-check
- With a newly reset database, run tests. 
- Using the application from the Admin user perspective:
1. On the Admin User Profile submit a few background checks and make sure they are appearing as they should
2. Try and delete one of the background checks that should now be in the history
3. Make sure the Toast appears in the bottom right confirming that the background check was deleted 
4. Reload the page to make sure the background check stays gone and check the database to make sure it is removed from there as well 
5. Repeat on a normal students User Profile and Student Staff User Profile.
